packages/cfd/src/Components/cfd-personal-details-form.tsx

```diff
@@ -217,24 +225,44 @@ const CFDPersonalDetailsForm = ({
     form_error,
 }: TCFDPersonalDetailsFormProps) => {
     const account_opening_reason = getAccountOpeningReasonList();
+    const { jurisdiction_selected_shortcode } = useCfdStore();
+    const {
+        client: { account_settings },
+    } = useStore();
+
+    const residence = residence_list?.find(item => item.text === account_settings?.residence);
+    const { data, isLoading: is_landing_company_details_loading } = useLandingCompanyDetails({
+        // @ts-expect-error jurisdiction_selected_shortcode type should be updated in cfd-store types
+        landing_company_details: jurisdiction_selected_shortcode,
+        country: residence?.value,
```

Comment:
Man this might hit rate limit. You're combining 2 different lifecycles here

Reply:
I've tested multiple times and never once I encountered any issue. Even QA tested this with the BE changes as this card needed an additional change from BE. 
Would you please suggest a better solution to this?

Reply:
Since cfd package is not apt for react query yet. I think its better to go with legacy first till @hamza-deriv does his refactor. But its @hamza-deriv bhai's call.

Reply:
This hook is being called from `@deriv/hooks` and if I recall it was mentioned by @yashim-deriv that these hooks rely on `stores` under the hood and they will be in sync within the execution flow. Please correct me if I'm missing something.

Reply:
You are still using useQuery from from api package. We should be sticking to one lifecycle. Plus your hook types are coming straight from api. Meanwhile in the store lifecycle as we see the types are coming from cfd-store-types.

Reply:
It just feels a bit messy to me to keep track of 2 different things.

Reply:
Would you suggest using WS directly in the component?

Reply:
I think this should be fine @thisyahlen-deriv. This is from the hooks package - `@deriv/hooks`, which should still sync up with the MobX stores